scvi/module/_mrdeconv.py

```diff
@@ -140,7 +140,7 @@ def _get_inference_input(self, tensors):
 
     def _get_generative_input(self, tensors, inference_outputs):
         x = tensors[REGISTRY_KEYS.X_KEY]
-        ind_x = tensors[REGISTRY_KEYS.X_KEY].long()
+        ind_x = tensors[REGISTRY_KEYS.INDICES_KEY].long().ravel()
```

Comment:
FYI @jjhong922 this was the big change, all the other changes are to prevent this from happening again